pkg/util/util_test.go

```
@@ -331,3 +332,79 @@ func TestConvertToSingleDocs(t *testing.T) {
 		})
 	}
 }
+
+func Test_ReplicatedAPIEndpoint(t *testing.T) {
+	tests := []struct {
+		name        string
+		license     *kotsv1beta1.License
+		isEmbedded  bool
+		envEndpoint string
+		want        string
+		wantError   bool
+	}{
+		{
+			name: "license with endpoint",
+			license: &kotsv1beta1.License{
+				Spec: kotsv1beta1.LicenseSpec{
+					Endpoint: "https://replicated.app",
+				},
+			},
+			isEmbedded: false,
+			want:       "https://replicated.app",
+		},
+		{
+			name: "license with endpoint including port",
+			license: &kotsv1beta1.License{
+				Spec: kotsv1beta1.LicenseSpec{
+					Endpoint: "https://replicated.app:8443",
+				},
+			},
+			isEmbedded: false,
+			want:       "https://replicated.app:8443",
+		},
+		{
+			name:       "no license uses default endpoint",
+			license:    nil,
+			isEmbedded: false,
+			want:       "https://replicated.app",
+		},
+		{
+			name:        "embedded cluster with env endpoint",
+			license:     nil,
+			isEmbedded:  true,
+			envEndpoint: "https://replicated.app",
+			want:        "https://replicated.app",
+		},
+		{
+			name:       "embedded cluster without env endpoint",
+			license:    nil,
+			isEmbedded: true,
+			wantError:  true,
+		},
```

```suggestion
		{
			name:       "embedded cluster without env endpoint",
			license:    nil,
			isEmbedded: true,
			wantError:  true,
		},
		{
			name:       "embedded cluster without env endpoint but with license",
			license: &kotsv1beta1.License{
				Spec: kotsv1beta1.LicenseSpec{
					Endpoint: "https://replicated.app:8443",
				},
			},
			isEmbedded: true,
			want: "https://replicated.app:8443",
		},
```

---

that should return an error